### ISSUE TITLE
Read correction jobs failed

### ISSUE THREAD

**Issue by kevlim83** [marbl/canu#272]
Hi,

I am at this stage of the correction process where the pipeline should be generating the correction layouts and generate the corrected reads.

However, the error logs in the canu-scripts folder indicate that all the jobs has failed. I tried to look for more clues in the various folders but could not find why the job failed.

This is a truncated copy of the error log:
```
--BEGIN CORRECTION
--
-- Set corMinCoverage=4 based on read coverage of 142.
-- Using overlaps no worse than 0.30 fraction error for correcting reads (from corErrorRate parameter).
--
-- 512 read correction jobs failed:
--   job /home/users/nus/gmsljk/assembly/canu_d1_100_sen/correction/2-correction/correction_outputs/0001.fasta FAILED.
--   job /home/users/nus/gmsljk/assembly/canu_d1_100_sen/correction/2-correction/correction_outputs/0002.fasta FAILED.
--   job /home/users/nus/gmsljk/assembly/canu_d1_100_sen/correction/2-correction/correction_outputs/0003.fasta FAILED.
--   job /home/users/nus/gmsljk/assembly/canu_d1_100_sen/correction/2-correction/correction_outputs/0004.fasta FAILED.
--   job /home/users/nus/gmsljk/assembly/canu_d1_100_sen/correction/2-correction/correction_outputs/0005.fasta FAILED.
```

In the 2-correction folder, the last thing that was written was the correctReads.sh file.

I have previously ran a successful correction using default settings, the only thing different that I am trying out now is to set the sensitivity to high before meeting this problem.

Please let me know if you have any ideas what went wrong or can point me in the right direction to get this run sorted.

Thanks

**Comment by brianwalenz** [comment]
Not much to go on there.  Are there no `*.err` files in the `2-correction` or `2-correction/correction_outputs` directories?  Are there any files at all in `correction_outputs`?

Try running correctReads.sh manually - `./correctReads.sh 1` will run the first job, etc.  Some errors might be logged to the screen, some might be logged to `correction_outputs/0001.err`.

**Comment by kevlim83** [comment]
Thanks. It worked without errors and the corrected reads are dumped into the fasta.gz file.

But I just wonder what happened to the pipeline that caused it to break here? Previously, there used to be another file correctReads.jobSubmit.sh that will automatically do its job.

**Comment by brianwalenz** [comment]
I'll guess memory contention with some other job.  The jobSubmit.sh files should still exist; they do in my recent grid-based runs.

**Comment by devenderarora** [comment]
Dear skoren,
I am getting the same error: 
correction/2-correction/correction_outputs/0463.fasta FAILED.
I am running the canu assembler on pig pac biodata and when the same data I ran earlier using genome size 2700m it completed successfully but when we decreased the genome size to 2500m we are getting this error. I tried running manually but it again failed to execute. Any help in this regards.

Later I tried with ./correctReads.sh 0463 but end with the error again.

Thanks

**Comment by skoren** [comment]
It's unlikely the errors are related since this issue is from a much older version of Canu. Correction won't be affected by a genome of 2.5g vs 2.7gb (pretty much the whole assembly wouldn't be affected by 2.5 vs 2.7gb genome size so if that's the only change between your two runs I would suggest just using the 2.7 result) so I expect this is a grid resource or timeout issue.  

Post the log from the failing job (```correction/2-correction/*463*``` and ```correction/2-correction/correction_outputs/*463*err```) and any logging from the grid you have for that job.

**Comment by devenderarora** [comment]
We observed assembly with genomic size 2700mb was 100% matched with the standard genome available in database. Which seems to be unlikely. Hence, we try to reduced the genome size to again cross check. I recently reinstalled the canu using bioconda and read somewhere we have less control  when installed canu using it. So change our strategy and re-run with standard installation. Till now it is working smoothly.
